File: interview/fizzbuzz.java
// Classic FizzBuzz implementation
class FizzBuzz {
  private final String NEWLINE = "\n";
  public String invoke (int maxNumber) {
    String buffer = "";
    for (int currentNumber = 1; currentNumber <= maxNumber; currentNumber++) {
      if (isMultipleOf(currentNumber, 5) && isMultipleOf(currentNumber, 3)) {
        buffer += "fizzbuzz" + NEWLINE;
      } else if (isMultipleOf(currentNumber, 3)) {
        buffer += "fizz" + NEWLINE;
      } else if (isMultipleOf(currentNumber, 5)) {
        buffer += "buzz" + NEWLINE;
      } else {
        buffer += currentNumber + NEWLINE;
      }
    }
    return buffer;
  }
  private boolean isMultipleOf(int number, int modNumber) {
    return number % modNumber == 0;
  }
  public static void main(String[] args) {
    FizzBuzz fb = new FizzBuzz();
    System.out.println(fb.invoke(100));
  }
}

File: interview/fizzbuzz.js
// from here
// https://github.com/blakeembrey/code-problems/tree/master/shortest-fizz-buzz
// Write a program that prints (to STDOUT) the numbers from 1 to 100. But for
// multiples of three print “Fizz” instead of the number and for the multiples
// of five print “Buzz”. For numbers which are multiples of both three and five
// print “FizzBuzz”.


// Do is as short as possible
//
function fizzBuzz(low, high) {
  low--; while (++low <= high) console.log( ( low % 3 ? '' :'Fizz') + ( low % 5? '' : 'Buzz') || low);
}

fizzBuzz(1,100);
